napari/_tests/test_viewer.py

```diff
@@ -357,3 +357,16 @@ def test_current_viewer(make_napari_viewer):
 
     assert current_viewer() is viewer1
     assert current_viewer() is not viewer2
+
+
+def test_reset_empty(make_napari_viewer):
+    """Test that resetting an empty viewer doesn't crash"""
+    viewer = make_napari_viewer()
```

Comment:
As `reset` is defined in `EventedModel` and the bug this reproduces should occur with `ViewerModel`, I think we should test that instead of the heavier full viewer.

Reply:
Ah wait, sorry, maybe `ViewerModel` is not enough - I could reproduce with it on `main`. Ignore if that's true.

Reply:
I do not know. I could not reproduce the error using `ViewerModel`. I just want to fix this bug before release.

Reply:
Agreed - I cannot reproduce it with `ViewerModel` either. There must be something about `Viewer` that's causing the issue. Not sure what that is, but I understand the changes here enough to approve.